### ISSUE TITLE
Bug when running "flutter doctor"

### ISSUE THREAD

**Issue by theunfactoringguru** [flutter/flutter#121354]
I found a bug on Linux systems. Specifically using Ubuntu 22.10

when running
```bash
$ flutter doctor
```

I get the following output:
```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.5, on Ubuntu 22.10 5.19.0-31-generic, locale
    en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[!] Android Studio (version 2022.1)
    ✗ Unable to find bundled Java version.
[✓] VS Code (version 1.75.1)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.

```
but when running Android applications, the application runs smoothly on my device and on emulators with no sort of errors.

I think this is a bug on the stable Snap release for flutter.

DISCLAIMER: This command ran on an Ubuntu 22.10 terminal. Also I am using the Snap flutter --classic release from Snapcraft.

**Comment by moslite** [comment]
- When you are already at the Android Studio installation root folder, you will see two folders: "jbr" and "jre".
- Copy all the files inside the "jbr" folder to the "jre" folder, then try it again.

**Comment by ciscofu** [comment]
> * When you are already at the Android Studio installation root folder, you will see two folders: "jbr" and "jre".
> * Copy all the files inside the "jbr" folder to the "jre" folder, then try it again.

Copying jbr to jre fixes the problem with flutter doctor, but it isn't a good solution. Android Studio properly recognizes it in the jbr folder. Copying the folder uses disk space unnecessarily, and since Android Studio isn't aware of the folder, the new jre folder won't be updated. It is likely that eventually (after Android Studio upgrades/patches) flutter doctor will report that the jre version is not correct for Android Studio, even though everything is fine because the jbr directory has been updated. Creating a symlink for jre that points to jbr is a better workaround (ln -s jbr jre) which avoids these problems.

That said, I agree with the bug report that it is something that should be fixed rather than worked around on each installation. The ideal fix is to have flutter doctor recognize the jbr folder as an alternate location for the JRE.

**Comment by gnprice** [comment]
This looks like another report of #106674.

**Comment by darshankawar** [comment]
@theunfactoringguru 
The fix for this is already available on latest master as part of https://github.com/flutter/flutter/issues/106674 but not in stable yet. There's a hotfix request raised by @gnprice https://github.com/flutter/flutter/issues/121505 which when approved and merged, the fix will be available in next hotfix stable version.
I suggest you to follow-up in cp issue for further updates. Will close this as a duplicate. If anybody disagrees, write in comments and I'll reopen it.

**Comment by JasonChiu-dev** [comment]
> @theunfactoringguru The fix for this is already available on latest master as part of #106674 but not in stable yet. There's a hotfix request raised by @gnprice #121505 which when approved and merged, the fix will be available in next hotfix stable version. I suggest you to follow-up in cp issue for further updates. Will close this as a duplicate. If anybody disagrees, write in comments and I'll reopen it.


Before the fix in next Flutter hotfix stable version, here is the solution for

**MacBook M1**:

Step 1: Change directory to the subdirectory (Contents) of the Android Studio forder:

`cd /Applications/Android\ Studio.app/Contents/`

Step 2: Create a symbolic link named jre and point it to jbr:

`ln -s jbr jre`

**Comment by theunfactoringguru** [comment]
Yeah the simbolic link worked with the issue

On Linux if using Jetbrains Toolchain

```bash
# the directory "221.6008.2211.9514443" may be unique to others I do not know
$ cd .local/share/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9514443/
$ ln -s jbr jre
```
